src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = addrs[i].MailingStreet__c;
+            contactsAddrs.contacts[i].MailingCity = addrs[i].MailingCity__c;
+            contactsAddrs.contacts[i].MailingState = addrs[i].MailingState__c;
+            contactsAddrs.contacts[i].MailingPostalCode = addrs[i].MailingPostalCode__c;
+            contactsAddrs.contacts[i].MailingCountry = addrs[i].MailingCountry__c;
+        }        
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.contains('New Street'));
+            system.assert(contact.MailingCity.contains('New City'));
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);
+        }
+        
+        // verify the new addresses still are Default and we don't have too many
+        List<Address__c> ListAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Contact__c, Latest_Start_Date__c, 
+        Latest_End_Date__c from Address__c];
+        system.assertEquals(2 * 2, ListAddr.size());
+        for (Address__c addr : ListAddr) {
+            boolean isNew = (addr.MailingStreet__c.contains('New Street'));
+            system.assertEquals(isNew, addr.Default_Address__c);
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Insert Contact w/ multiline street address
+    verify: new default Address created and the street address is broken up into street and street2
+    **********************************************************************************************************/            
+    @isTest
+    public static void insertMultilineStreetAddress() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Contacts_Addresses_Enabled__c = true));
+    
+        List<Contact> contacts = UTIL_UnitTestData_TEST.CreateMultipleTestContacts(2);
+        for (Contact contact : contacts) {
+            contact.MailingStreet = 'new street\r\nsecond line';
+            contact.MailingCity = 'new city';
+        }
+        Test.startTest();
+        insert contacts; 
+        Test.stopTest();       
+        
+        // verify the Address objects split street.
+        List<Address__c> ListAddr = [select Id, MailingStreet__c, MailingStreet2__c, Formula_MailingStreetAddress__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.